File: Main/brain/Headlights.js
// Headlights.js - Javascript interface for controlling the headlights of Poseidon
//
// Created by Kevin Burgon on 02/01/2016

function Headlights(cmdOut) {

	this.toggleLights = function() {
		child.stdin.write("switchLights");
	}
}

module.exports = Headlights;

File: Main/Peripherals/src/Drivers/Headlights.cpp
// Headlights.cpp - functions for toggling the headlights on Poseidon
//
// Created by Kevin Burgon on 02/01/2016

#include "Headlights.h"

Headlights::Headlights(Serial& serial) : serial_(serial)
{
	areOn = false;
}

void Headlights::switchLights()
{
	if (areOn)
	{

		serial_.writeByte(0x9);
		serial_.writeByte(0x0);
		areOn = false;
	}
	else
	{
		serial_.writeByte(0x9);
		serial_.writeByte(0x1);
		areOn = true;
	}
}
